src/opensynth/models/faraday/gaussian_mixture/fit_kmeans.py

```diff
@@ -33,36 +33,6 @@ def fit_kmeans(
     Returns:
         torch.Tensor: k-means centroids
     """
-    # Initiate K-means model
-    # kmeans_model_ = KMeansModel(
-    #     num_clusters=num_components, num_features=input_dim
-    # )
-
-    # # Use uniform distribution to get initial centroids
-    # init_module = KmeansRandomInitLightningModule(
-    #     kmeans_model_, vae_module, num_components, input_dim
-    # )
-    # trainer = pl.Trainer(
-    #     max_epochs=1, accelerator=accelerator, devices=devices, logger=logger
-    # )  # setting initial values - run for 1 epoch
-    # trainer.fit(init_module, data)
-
-    # # Fit K-means
-    # kmeans_module = KMeansLightningModule(
-    #     kmeans_model_,
-    #     vae_module,
-    #     num_components,
-    #     input_dim,
-    #     convergence_tolerance,
-    # )
-    # trainer = pl.Trainer(
-    #     max_epochs=max_epochs,
-    #     accelerator=accelerator,
-    #     devices=devices,
-    #     logger=logger,
-    # )
-    # trainer.fit(kmeans_module, data)
-
     # TODO : perform k-means on a random subsample of the training dataset to
     # speed up convergence. This is necessary when working with large datasets.
     kmeans_model_ = KMeans(n_clusters=num_components)
```

Comment:
Lines 40-44 can be replaced by calling the function `encode_data_for_gmm` from `prepare_gmm_input.py`. (Except you would still have to  `.detach().numpy()` here as this is not done in  `encode_data_for_gmm`). Note, do not use the `prepare_data_for_model` function as this will expand the dataset based on the weights which we do not want to do for k-means training (only GMM).